internal/impl/ollama/base_processor.go

```diff
@@ -55,22 +65,109 @@ func (co *commandOutput) Write(b []byte) (int, error) {
 
 type key int
 
-const loggerKey key = key(1)
+const (
+	configKey key = key(1)
+)
+
+type runOllamaConfig struct {
+	logger      *service.Logger
+	fs          *service.FS
+	cacheDir    string
+	downloadURL string
+}
+
+func (c *runOllamaConfig) lookPath(file string) (string, error) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		return "", errors.ErrUnsupported
+	}
+	path := os.Getenv("PATH")
+	for _, dir := range filepath.SplitList(path) {
+		if !filepath.IsAbs(path) {
+			continue
+		}
+		path := filepath.Join(dir, file)
+		p, err := c.fs.Stat(path)
+		if err != nil || p.IsDir() {
+			continue
+		}
+		// Check that the file is executable
+		if err := syscall.Access(path, 0x1); err == nil {
+			return path, nil
+		}
+	}
+	return "", exec.ErrNotFound
+}
+
+func (c *runOllamaConfig) downloadOllama(ctx context.Context, path string) error {
+	var url string
+	const baseURL string = "https://github.com/ollama/ollama/releases/download/v0.3.1/ollama"
```

Comment:
Should we consider exposing the version as a config parameter?

Reply:
Oh probably since this is already old 😄

Reply:
Well you can already provide a override URL to download from whereever, so I think that is a good enough workaround for now